protos/dcs/common/v0/common.proto

```diff
@@ -388,3 +388,26 @@ message Vector {
   double y = 2;
   double z = 3;
 }
+
+/**
+ * An instance of a contact in a DCS AI controller's detection table
+ *
+ * This is a target that the AI controller has detected and is actively tracking
+ *
+ */
+message Contact {
```

Comment:
I think a better way of doing this would be to use `oneof`

Then we could have

```proto
message Contact {
  message BasicInformation {
    // The contact's name
    string name = 1;
    // The DCS generated ID
    uint32 id = 2;
    // The type of unit the contact is
    uint32 type = 3;
  }

  oneof details {
    BasicInformation basic_info = 1;
    Unit unit = 2;
    Weapon weapon = 3;
  }
  bool is_visible = 4; // I am not entirely sure what this means. If it is not visible then it would not be in the list, no?
  bool sitance = 5;
}
```

Which removes all duplication

Reply:
Agreed,

Committed changes to implement this. 

As for the comment about is_visible...it's documented in the DCS API as if the unit can see the target or not, and it doesn't really go into much detail for that. I don't actually fully understand what it represents, but I included it because it was included in getDetectedTargets().

Reply:
I think it might be useful for targets that are detected via datalink.